examples/40_advanced/example_early_stopping_and_callbacks.py

```diff
@@ -0,0 +1,75 @@
+"""
+============================
+Early stopping and Callbacks
+============================
+
+The example below shows how we can use the ``get_trials_callback`` parameter of
+auto-sklearn to implement an early-stopping mechanism through a callback.
+
+These callbacks give access to the result of each model + hyperparameter configuration
+optimized by SMAC, the underlying optimizer for autosklearn. By checking the cost of
+a result, we can implement a simple yet effective early stopping mechanism!
+
+Do note however, this does not provide any access to the ensembles that autosklearn
+produces, only the individual models. You may wish to perform a more sophisticated
+early stopping mechanism such that there are enough good models for autosklearn to build
+and ensemble with. This is here to provide a simple example.
+"""
+from pprint import pprint
+
+import sklearn.datasets
+import sklearn.metrics
+
+import autosklearn.classification
+
+from smac.optimizer.smbo import SMBO
+from smac.runhistory.runhistory import RunInfo, RunValue
+
+
+############################################################################
+# Build and fit a classifier
+# ==========================
+def callback(
+    smbo: SMBO,
+    run_info: RunInfo,
+    result: RunValue,
+    time_left: float,
+) -> bool:
```

Comment:
Maybe a brief description what this bool means would be good. Also, the signature doesn't match the return value at the moment (return value is `bool | None`).

Reply:
Good catch